fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift

```diff
@@ -3,22 +3,61 @@ import UIKit
 final class NftDetailsInteractor {
     // MARK: - Private properties
 
+    private let nftFetchingService: NFTFetchingServiceProtocol
     private weak var output: NftDetailsInteractorOutput?
     private var nft: NFT?
+    private var isReady: Bool = false
 
-    init(nft: NFT?) {
+    init(nft: NFT?, nftFetchingService: NFTFetchingServiceProtocol) {
         self.nft = nft
+        self.nftFetchingService = nftFetchingService
     }
 }
 
 // MARK: - NftDetailsInteractorInput
 
 extension NftDetailsInteractor: NftDetailsInteractorInput {
+    func initialSetup() {
+        let wasReady = isReady
+        isReady = true
+
+        if !wasReady {
+            fetchData()
+        }
+    }
+
     func setup(with output: NftDetailsInteractorOutput) {
         self.output = output
 
         if let nft = nft {
             output.didReceive(nft: nft)
         }
     }
+
+    func fetchData() {
+        guard isReady else {
+            return
+        }
+
+        isReady = false
+
+        Task {
+            do {
```

Comment:
catch the error please

Reply:
you should change isReady to false if error caught?

Reply:
There is nothing to catch, if we didn't got owners user shouldn't know about that.